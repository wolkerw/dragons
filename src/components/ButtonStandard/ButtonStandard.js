import React from "react";
import PropTypes from "prop-types";

function ButtonStandard(props) {
    const {
        text,
        textId,
        textClass,
        className,
        type,
        disabled,
        onClick,
        isLoading
    } = props;

    const showText =
        textId === undefined && text === undefined ? (
            ""
        ) : text !== undefined && textId === undefined ? (
            <span className={textClass}>{text}</span>
        ) : (
            <span>{textId}</span>
        );

    return (
        <button
            type={type ? type : "buttom"}
            className={className + (isLoading ? " loading" : "")}
            disabled={isLoading ? true : disabled}
            onClick={isLoading ? null : onClick}
        >
            {isLoading ? <p>Carregando</p> : null}
            {showText}
        </button>
    );
}
export default ButtonStandard;

ButtonStandard.propTypes = {
    text: PropTypes.string,
    textId: PropTypes.string,
    textClass: PropTypes.string,
    disabled: PropTypes.bool,
    textValues: PropTypes.object,
    className: PropTypes.string,
    onClick: PropTypes.func
};
