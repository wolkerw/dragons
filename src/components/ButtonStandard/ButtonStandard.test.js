import React from "react";
import ButtonStandard from "./ButtonStandard";
import { IntlProvider } from "react-intl";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
// import messages from "../../assets/i18n";

const country = "pt-BR";

configure({ adapter: new Adapter() });

describe("ButtonStandard Component", () => {
    const component = (
        <IntlProvider
            locale={country}
            // key={country}
            // messages={messages[{ country }]}
            // defaultLocale={country}
        >
            <ButtonStandard
                textId="Entrar"
                className="btn btn-primary"
                onClick={() => {
                    console.log("FUNCIONOU");
                }}
            />
        </IntlProvider>
    );
    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is clickable", () => {
        const wrapper = mount(component);
        expect(wrapper.find(ButtonStandard).simulate("click"));
    });
});
