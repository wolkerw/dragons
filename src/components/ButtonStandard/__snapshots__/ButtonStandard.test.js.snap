// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonStandard Component matches the snapshot 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="buttom"
>
  <span>
    Entrar
  </span>
</button>
`;
